internal/pkg/diagnostics/diagnostics.go

```diff
@@ -95,6 +95,29 @@ func GlobalHooks() Hooks {
 				return fileBytes
 			},
 		},
+		{
+			Name:        "environment",
+			Filename:    "environment.yml",
+			Description: "Environment variables",
+			ContentType: "application/yaml",
+			Hook: func(_ context.Context) []byte {
+				envMap := map[string]any{}
+				for _, e := range os.Environ() {
+					pair := strings.SplitN(e, "=", 2)
+					envMap[pair[0]] = pair[1]
+				}
+				var errOut bytes.Buffer
+				redacted := Redact(envMap, &errOut)
+				if errOut.Len() > 0 {
+					return errOut.Bytes()
+				}
+				out, err := yaml.Marshal(redacted)
```

Comment:
Can you add a test that this works and in particular that the result is correctly redacted?

I see the code doing it, but I want it to keep working if the code changes later.